### PR TITLE
Submitting assignment 4

### PR DESCRIPTION
**Pull request by Rulo401**

### CODE REVIEW

File: Assignment4/Rulo401-190243/task06.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1hSv61ejXESrp34CUz1PqZCsoA43tzc8y
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/Rulo401-190243/task06.py]: won't work in python. This is notebook magic

File: Assignment4/Rulo401-190243/task07.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  people = g.triples((None, RDF.type, s))
+  for s,p,o in people:
+    print(s)
+
+
+q2 = prepareQuery('''
+  SELECT DISTINCT ?Person WHERE { 
+    {?Person RDF:type ns:Person} UNION
+    {?Person RDF:type ?Class.
+    ?Class  RDFS:subClassOf ns:Person}
```
Comment (dgarijo) [inline Assignment4/Rulo401-190243/task07.py]: you need to have ALL the subclasses. Hint you may use chained properties with "*" for 0..n repetitions

File: Assignment4/Rulo401-190243/task07.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  SELECT ?Person (GROUP_CONCAT(?Property ; separator=" | ") as ?Properties) WHERE{
+    ?Person ?Property ?Value.
+    {
+        SELECT DISTINCT ?Person WHERE { 
+        {?Person RDF:type ns:Person} UNION
+        {?Person RDF:type ?Class.
+        ?Class  RDFS:subClassOf ns:Person}
+        }
+    }
+  }GROUP BY ?Person
```
Comment (dgarijo) [inline Assignment4/Rulo401-190243/task07.py:104]: nicely done, but it has the same issue as above. We need all the subclass hierarchy, not 1 level

File: Assignment4/Rulo401-190243/task09.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+rs = []
+
+for r in g1.query(q):
+  rs.append((r.Person, r.Given, r.Family))
+
+for r2 in g2.query(q):
+  for r1 in rs:
+    if(r1[1] == r2.Given and r1[2] == r2.Family):
+      g3.add((r1[0], OWL.sameAs, r2.Person))
+      g3.add((r2.Person, OWL.sameAs, r1[0]))
```
Comment (dgarijo) [inline Assignment4/Rulo401-190243/task09.py]: sameAs is reflexive, so you only need to assert it in 1 direction

**Review by dgarijo — CHANGES_REQUESTED** [review]
Small changes requested

**Comment by Rulo401** [comment]
@dgarijo Requested changes are done

**Review by dgarijo — APPROVED** [review]

**Pull request merged**